### PR TITLE
[CARBONDATA-3842] Fix incorrect results on mv with limit (Missed code during mv refcatory)

### PR DESCRIPTION
**Pull request by Indhumathi27**
### Why is this PR needed?
 Already issue fixed in [PR-3652](https://github.com/apache/carbondata/pull/3651). Missed code during mv code refactory
 
 ### What changes were proposed in this PR?
Copy subsume Flag and FlagSpec to subsumerPlan while rewriting with summarydatasets.
Update the flagSpec as per the mv attributes and copy to relation.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1403/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3127/

**Comment by akashrn5** [comment]
@Indhumathi27 why the test cases didnt catch this. If we do not have test cases, please add a test case for the limit case.

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala [hunk 1]
```diff
@@ -610,10 +610,10 @@ class TestAllOperationsOnMV extends QueryTest with BeforeAndAfterEach {
     sql("drop materialized view if exists mv1")
     sql("create materialized view mv1  as select a.name,a.price from maintable a")
     var dataFrame = sql("select a.name,a.price from maintable a limit 1")
-    assert(dataFrame.count() == 1)
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala:613]: why this change? because count is also an action, it should also give 1

Comment (Indhumathi27) [reply integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala:613]: In this Case, count was returning 1 only, but df.show() is giving all results. In cluster also, limit query gives all rows. Because of this, testcase didnt catch this issue. That's why changed to df.collect.length

Comment (Indhumathi27) [reply integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala:613]: Here, count API is not applying collect rows on SPARK PLAN. When i checked it, plan was on main table

**Comment by Indhumathi27** [comment]
retest this please

File: integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala (hunk 1 above)
Comment (Indhumathi27) [reply integration/spark/src/test/scala/org/apache/carbondata/view/rewrite/TestAllOperationsOnMV.scala:613]: @akashrn5 Please help to review and merge

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1563/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3300/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1578/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3316/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3335/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1595/

**Comment by akashrn5** [comment]
LGTM
